### PR TITLE
Upgrade tooling and dependencies e.g. pnpm, vitest, xstate

### PR DESCRIPTION
**Pull request by plmercereau**
- upgrade to pnpm 7.9.1 (latest version accepted by Vercel)
- lock pnpm version in `package.json` so we (hopefully) stop re-generating the lock file on every PR 
- Bump xstate and remove the [`predictableActionArguments` warning](https://github.com/nhost/nhost/issues/882)
- Simplify typings in tests
- Use `vi.stubGlobal` to [use Vitest 0.24](https://github.com/nhost/nhost/issues/921)
- Bump other tooling dependencies e.g. vite, typedoc, eslint

### CODE REVIEW

**Review by elitan — APPROVED** [review]

File: packages/vue/src/useSignUpEmailPassword.ts [hunk 1]
```diff
@@ -1,4 +1,4 @@
-import { computed, ToRefs, unref } from 'vue'
```
Comment (szilarddoro) [inline packages/vue/src/useSignUpEmailPassword.ts:1]: What do you think about enabling this in VSCode settings on a repository level through `.vscode`? I found it very useful in almost all cases. The only problem you can face by enabling this is that commenting out code blocks and saving the file results in lost imports, but it's very easy to restore them. Generally it helps keeping imports clean.

```json
{
  "editor.codeActionsOnSave": {
    "source.organizeImports": true
  }
}
```

Comment (plmercereau) [reply packages/vue/src/useSignUpEmailPassword.ts:1]: Yes. I also like to use the `recommendations` field.
And we'll add this when using JSON schemas:
```json
{
  "yaml.schemas": {
    "./packages/config-schema/schema.json": "nhost/config.yaml"
  }
}
```

Comment (plmercereau) [reply packages/vue/src/useSignUpEmailPassword.ts:1]: anyway I added the `.vscode` directory

File: package.json [hunk 2]
```diff
@@ -23,7 +23,7 @@
     "clean:all": "pnpm clean && rm -rf ./{{packages,examples}/*,docs}/{.nhost,node_modules} node_modules",
     "clean": "rm -rf ./{{packages,examples}/*,docs}/{dist,umd,.next,.turbo,coverage}",
     "ci:version": "changeset version && pnpm install --frozen-lockfile false",
-    "coverage": "pnpm run test --coverage",
+    "coverage": "pnpm run test -- --coverage",
```
Comment (nunopato) [inline package.json:26]: is this change on purpose? :eyes:
